### ISSUE TITLE
Incorrect fetchSpec for non-github-hosted git+ssh :user pattern

### ISSUE THREAD

**Issue by moos** [npm/npm-package-arg#22]
```
> npa('git+ssh://git@github.not:foo/bar')
{ type: 'git',
  registry: undefined,
  where: undefined,
  raw: 'git+ssh://git@github.not:foo/bar',
  name: undefined,
  escapedName: undefined,
  scope: undefined,
  rawSpec: 'git+ssh://git@github.not:foo/bar',
  saveSpec: 'git+ssh://git@github.not:foo/bar',
  fetchSpec: 'ssh://git@github.not/:foo/bar',
  gitRange: undefined,
  gitCommittish: 'master',
  hosted: undefined }
```
Note incorrect **/:foo** in `fetchSpec`.

Versions tested: 5.0.0 & 5.0.1.

**Comment by moos** [comment]
Incidentally, github.com-hosted works fine
```
> npa('git+ssh://git@github.com:foo/bar')
{ type: 'git',
  registry: undefined,
  where: undefined,
  raw: 'git+ssh://git@github.com:foo/bar',
  name: undefined,
  escapedName: undefined,
  scope: undefined,
  rawSpec: 'git+ssh://git@github.com:foo/bar',
  saveSpec: 'git+ssh://git@github.com/foo/bar.git',
  fetchSpec: 'ssh://git@github.com/foo/bar.git',
  gitRange: undefined,
  gitCommittish: 'master',
  hosted: { ... }
```
Note correct `fetchSpec`.

**Comment by iarna** [comment]
This was fixed with v5.1.2 and 8d26082:
```
> require('npa')('git+ssh://git@github.not:foo/bar')
{ type: 'git',
  registry: undefined,
  where: undefined,
  raw: 'git+ssh://git@github.not:foo/bar',
  name: undefined,
  escapedName: undefined,
  scope: undefined,
  rawSpec: 'git+ssh://git@github.not:foo/bar',
  saveSpec: 'git+ssh://git@github.not:foo/bar',
  fetchSpec: 'git@github.not:foo/bar',
  gitRange: undefined,
  gitCommittish: 'master',
  hosted: undefined }
```
